app/queries/vendor_api_request_summary_query.rb

```diff
@@ -0,0 +1,78 @@
+class VendorAPIRequestSummaryQuery
+  ALL_TIME = 'all_time'.freeze
+  LAST_WEEK = 'last_week'.freeze
+  LAST_MONTH = 'last_month'.freeze
+
+  def initialize(sort_param = ALL_TIME)
+    @sort_param = sort_param
+  end
+
+  def call
+    ActiveRecord::Base.connection.exec_query(
+      count_sql,
+      'SQL',
+      [[nil, 1.week.ago.beginning_of_day], [nil, 1.month.ago.beginning_of_day]],
+    ).to_a
+  end
+
+private
+
+  def count_sql
+    "WITH vendor_api_request_counts AS (
+      SELECT
+        request_path,
+        jsonb_extract_path_text(jsonb_array_elements(response_body->'errors'), 'error') AS attribute,
```

Comment:
this will extract text from the `error` field nested in an array of hashes under the `errors` section in the response body. Let me know if there is a better way to query this.

Reply:
I can't think of a better way to achieve this with the current data structure.

However, as we already know that we are interested in the errors occuring for unprocessable requests maybe we should consider updating the VendorAPIRequest model to include an errors field, and the relevant worker to store the errors in that field as they occur, to make them easier to access for both development but potentially also by the analysts via blazer.

Reply:
That is a good idea. I think I'd like to tackle this in a further iterations as adding a new column now would mean we would need to backfill those values for existing fields, something that is beyond the scope of this PR.